src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -305,11 +335,13 @@ def create_widget(self, dashboard, widget_data, order):
         DashboardWidgetQuery.objects.bulk_create(new_queries)
 
     def update_widget(self, widget, data, order):
+        prev_layout = widget.detail.get("layout", None) if widget.detail else None
         widget.title = data.get("title", widget.title)
         widget.display_type = data.get("display_type", widget.display_type)
         widget.interval = data.get("interval", widget.interval)
         widget.widget_type = data.get("widget_type", widget.widget_type)
         widget.order = order
+        widget.detail = {"layout": data.get("layout", prev_layout)}
```

Comment:
Updating widget.detail might start to get a bit ugly as we add more keys to the `detail` dict. Wondering if we need access to the widget instance and have to get prev_layout? If we should do something like in the validate() of [DiscoverSavedQuery](https://github.com/getsentry/sentry/blob/03ee96982b9e86bce623f2c518f7ee6b1e9bf6ba/src/sentry/discover/endpoints/serializers.py#L225-L230) where we build the detail dict in the validate? 

What we have might be fine for now though.

Reply:
I agree that it's going to start looking messy when we start adding more fields.

This line is covered by the `test_update_without_specifying_layout_does_not_change_saved_layout` test. Now I'm wondering if this behaviour is too flexible. If we don't grab `prev_layout` then we could just default the layout to `None`.

At the time I was thinking of the behaviour not just from a UI usage, but a general use case and avoiding unintended destructive behaviour, seeing as how the rest of this function does something similar.

Reply:
We chatted about this on a call. In summary we decided to keep this as-is and conduct a separate discussion on the behaviour of endpoints. Our concerns here are that this is a PUT request for a dashboard, but this almost acts like a PATCH for widgets. It is slightly confusing what defaults we should have in a case of nested serializers like this.